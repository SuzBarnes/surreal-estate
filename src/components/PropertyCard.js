import React from "react";
import "../styles/propertycard.css"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBath, faBed } from "@fortawesome/free-solid-svg-icons";


const PropertyCard = ({ title, city, type, bedrooms, bathrooms, price, email }) => {
    const sendMail = () => {
        const mailto: string =
            `mailto:${email}?subject=Request viewing of ${title}`;
        window.location.href = mailto;
    }

    return (
        <div className="property-card">
            <div className="property-title"><h3>{title}</h3></div>
            <div className="property-city">{city}</div>
            <div className="property-type">{type}</div>
            <div className="property-bedrooms"> <FontAwesomeIcon icon={faBed} className="font-awesome" />{bedrooms}</div>
            <div className="property-bathrooms"><FontAwesomeIcon icon={faBath} className="font-awesome" />{bathrooms}</div>
            <div className="property-price">{price}</div>
            <div className="property-email">
                <button type="submit" className="property-button" onClick={sendMail}>Request Viewing</button>
            </div>
        </div >
    );
}


export default PropertyCard;