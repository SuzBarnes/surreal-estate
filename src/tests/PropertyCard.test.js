import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
    const propertyCard = {
        _id: 1,
        title: "mock title",
        type: "mock type",
        bathrooms: 1,
        bedrooms: 2,
        price: 100000,
        city: "mock city",
        email: "mock email",
    };

    const { asFragment } = render(<PropertyCard
        propertyCard={propertyCard}
    />)

    expect(asFragment()).toMatchSnapshot();

    it("displays the title, type, city and email correctly", () => {
        expect(propertyCard.title).toBe("mock title");
        expect(propertyCard.type).toBe("mock type");
        expect(propertyCard.city).toBe("mock city");
        expect(propertyCard.email).toBe("mock email");
    })

    it("displays the number of bedrooms, bathrooms, price as the correct numbers", () => {
        expect(propertyCard.bedrooms).toBe(2);
        expect(propertyCard.bathrooms).toBe(1);
        expect(propertyCard.price).toBe(100000);
    })

});