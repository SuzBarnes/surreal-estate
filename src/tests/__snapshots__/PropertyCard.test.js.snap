// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="property-title"
    >
      <h3 />
    </div>
    <div
      class="property-city"
    />
    <div
      class="property-type"
    />
    <div
      class="property-bedrooms"
    >
       
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bed font-awesome"
        data-icon="bed"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 32c17.7 0 32 14.3 32 32V320H288V160c0-17.7 14.3-32 32-32H544c53 0 96 43 96 96V448c0 17.7-14.3 32-32 32s-32-14.3-32-32V416H352 320 64v32c0 17.7-14.3 32-32 32s-32-14.3-32-32V64C0 46.3 14.3 32 32 32zM176 288c-44.2 0-80-35.8-80-80s35.8-80 80-80s80 35.8 80 80s-35.8 80-80 80z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="property-bathrooms"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bath font-awesome"
        data-icon="bath"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 77.3c0-7.3 5.9-13.3 13.3-13.3c3.5 0 6.9 1.4 9.4 3.9l14.9 14.9C130 91.8 128 101.7 128 112c0 19.9 7.2 38 19.2 52c-5.3 9.2-4 21.1 3.8 29c9.4 9.4 24.6 9.4 33.9 0L289 89c9.4-9.4 9.4-24.6 0-33.9c-7.9-7.9-19.8-9.1-29-3.8C246 39.2 227.9 32 208 32c-10.3 0-20.2 2-29.2 5.5L163.9 22.6C149.4 8.1 129.7 0 109.3 0C66.6 0 32 34.6 32 77.3V256c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H96V77.3zM32 352v16c0 28.4 12.4 54 32 71.6V480c0 17.7 14.3 32 32 32s32-14.3 32-32V464H384v16c0 17.7 14.3 32 32 32s32-14.3 32-32V439.6c19.6-17.6 32-43.1 32-71.6V352H32z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="property-price"
    />
    <div
      class="property-email"
    >
      <button
        class="property-button"
        type="submit"
      >
        Request Viewing
      </button>
    </div>
  </div>
</DocumentFragment>
`;
